pkg/statistics/handle/storage/save.go

```diff
@@ -129,7 +129,28 @@ func SaveAnalyzeResultToStorage(sctx sessionctx.Context,
 	// 1. Save mysql.stats_meta.
 	var rs sqlexec.RecordSet
 	// Lock this row to prevent writing of concurrent analyze.
-	rs, err = util.Exec(sctx, "select snapshot, count, modify_count from mysql.stats_meta where table_id = %? for update", tableID)
+	// Add a fake table ID to prevent the deadlock between the batch get and the point get.
+	//
+	// Deadlock Issue Explanation:
+	// We encountered a stable deadlock issue during the analyze process where the transaction
+	// ultimately failed and the error was propagated to the top layer. From the deadlock records
+	// in information_schema.deadlocks, it's confirmed that this is a non-retryable deadlock.
+	//
+	// Root Cause:
+	// The point get operation acquires locks in two separate phases: first the index lock, then the row lock.
+	// Meanwhile, another transaction doing a batch point get (e.g. during stats delta dump) acquires both locks
+	// together but in the opposite order (row then index). This creates a circular wait condition where:
+	// 1. Transaction 1 holds index lock and waits for row lock
+	// 2. Transaction 2 holds row lock and waits for index lock
+	// Since the locks are acquired in separate lockKeys calls, TiDB cannot detect and retry this deadlock.
+	//
+	// Deadlock Sequence:
+	// txn1: lockKeys on point get (index lock)
+	// txn2: lockKeys on batch point get (row lock) — waits for txn1 for index lock
+	// txn1: lockKeys on point get (row lock) — deadlock occurs here and it's not retryable
```

Comment:
@you06 Could you please help check the accuracy of the comment here? Thank you!

Reply:
LGTM